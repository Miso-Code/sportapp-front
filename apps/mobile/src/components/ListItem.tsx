import React from 'react'
import { StyleSheet } from 'react-native'
import { List, ListIconProps } from 'react-native-paper'

const LeftIcon = ({ icon, ...props }) => <List.Icon {...props} icon={icon} />
const RightIcon = (props) => <List.Icon {...props} icon='chevron-right' />

const ListItem: React.FC<{
	icon?: ListIconProps['icon']
	title: string
	onPress?: () => void
}> = ({ icon = 'folder', title, onPress = () => {} }) => {
	return (
		<List.Item
			testID='list-item'
			title={title}
			/* eslint-disable react/no-unstable-nested-components*/
			left={(props) => <LeftIcon {...props} icon={icon} key='leftIcon' />}
			right={(props) => <RightIcon {...props} key='rightIcon' />}
			/* eslint-enable react/no-unstable-nested-components*/
			onPress={onPress}
			style={styles.listItem}
		/>
	)
}

const styles = StyleSheet.create({
	listItem: {
		marginVertical: 5,
		borderColor: '#E0E0E0', // use theme colors
		borderWidth: 1,
		borderRadius: 5
	}
})

export default ListItem
