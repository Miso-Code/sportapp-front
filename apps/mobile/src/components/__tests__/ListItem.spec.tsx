import React from 'react'
import renderer, { ReactTestRenderer } from 'react-test-renderer'

import ListItem from '../ListItem'

describe('ListItem', () => {
	let component: ReactTestRenderer

	beforeEach(() => {
		component = renderer.create(
			<ListItem title='Item' onPress={() => {}} />
		)
	})

	afterEach(() => {
		component.unmount()
	})

	it('should render correctly', () => {
		expect(component.toJSON()).toMatchSnapshot()
	})

	it('should call the onPress function when the item is pressed', () => {
		const onPress = jest.fn()
		component.update(<ListItem title='Click me' onPress={onPress} />)
		component.root.findByProps({ testID: 'list-item' }).props.onPress()
		expect(onPress).toHaveBeenCalled()
	})

	it('should not call the onPress function when the button is pressed and the onPress prop is not provided', () => {
		const onPress = jest.fn()
		component.update(<ListItem title="Don't click me" />)
		component.root.findByProps({ testID: 'list-item' }).props.onPress()
		expect(onPress).not.toHaveBeenCalled()
	})
})
