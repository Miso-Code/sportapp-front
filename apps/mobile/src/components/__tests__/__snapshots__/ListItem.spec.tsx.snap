// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      false,
      [
        {
          "paddingRight": 24,
          "paddingVertical": 8,
        },
        {
          "borderColor": "#E0E0E0",
          "borderRadius": 5,
          "borderWidth": 1,
          "marginVertical": 5,
        },
      ],
    ]
  }
  testID="list-item"
>
  <View
    style={
      {
        "flexDirection": "row",
        "marginVertical": 6,
        "width": "100%",
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "alignSelf": "center",
            "marginLeft": 16,
            "marginRight": 0,
            "marginVertical": 0,
          },
        ]
      }
    />
    <View
      style={
        [
          {
            "paddingLeft": 16,
          },
          {
            "flexGrow": 1,
            "flexShrink": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
      testID="list-item-content"
    >
      <Text
        numberOfLines={1}
        selectable={false}
        style={
          [
            {
              "textAlign": "left",
            },
            {
              "color": "rgba(28, 27, 31, 1)",
              "fontFamily": "System",
              "fontWeight": "400",
              "letterSpacing": 0,
            },
            {
              "writingDirection": "ltr",
            },
            [
              {
                "fontSize": 16,
              },
              {
                "color": "rgba(28, 27, 31, 1)",
              },
              undefined,
            ],
          ]
        }
      >
        Item
      </Text>
    </View>
    <View
      pointerEvents="box-none"
      style={
        [
          {
            "alignItems": "center",
            "justifyContent": "center",
          },
          {
            "alignSelf": "center",
            "marginLeft": 16,
            "marginRight": 0,
            "marginVertical": 0,
          },
        ]
      }
    >
      <Text
        accessibilityElementsHidden={true}
        allowFontScaling={false}
        importantForAccessibility="no-hide-descendants"
        pointerEvents="none"
        selectable={false}
        style={
          [
            {
              "color": "rgba(73, 69, 79, 1)",
              "fontSize": 24,
            },
            [
              {
                "lineHeight": 24,
                "transform": [
                  {
                    "scaleX": 1,
                  },
                ],
              },
              {
                "backgroundColor": "transparent",
              },
            ],
            {
              "fontFamily": "material-community",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            {},
          ]
        }
      >
        ó°…‚
      </Text>
    </View>
  </View>
</View>
`;
